geode-core/src/main/java/org/apache/geode/internal/cache/BackupUtil.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache;
+
+import java.io.File;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Set;
+
+import org.apache.geode.admin.internal.AdminDistributedSystemImpl;
+import org.apache.geode.admin.internal.BackupDataStoreHelper;
+import org.apache.geode.admin.internal.BackupDataStoreResult;
+import org.apache.geode.cache.persistence.PersistentID;
+import org.apache.geode.distributed.internal.DM;
+import org.apache.geode.internal.i18n.LocalizedStrings;
+import org.apache.geode.admin.BackupStatus;
+import org.apache.geode.admin.internal.BackupStatusImpl;
+import org.apache.geode.management.ManagementException;
+
+public class BackupUtil {
+
+  private BackupUtil() {
+    // do not instantiate
+  }
+
+  public static BackupStatus backupAllMembers(DM dm, File targetDir, File baselineDir)
```

Comment:
Another refactor to do (it does not need to be done as part of this one) is to no longer use the deprecated admin package in the backup implementation. BackupStatus is the main offender here. But also consider moving the backup logic in admin.internal to a non-deprecated package.

Reply:
I would be happy to do that as part of this request. I started down that route, but ran into questions on how best to handling the fact that these classes are part of the current public API and therefore should not be removed outside of a major release (at least that is my understanding).